src/animation.rs

```diff
@@ -234,6 +236,26 @@ pub struct Action {
     pointer: froggy::Pointer<ActionData>,
 }
 
+impl PartialEq for Action {
+    fn eq(
+        &self,
+        other: &Action,
+    ) -> bool {
+        self.pointer == other.pointer
+    }
+}
+
+impl Eq for Action {}
+
+impl Hash for Action {
+    fn hash<H: Hasher>(
+        &self,
+        state: &mut H,
+    ) {
+        self.pointer.hash(state);
```

Comment:
Is there any way the value `self.pointer` can change? If not, then all is well.

Reply:
I don't think it's possible